chainimport/headers_import.go

```diff
@@ -0,0 +1,1814 @@
+package chainimport
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/neutrino/headerfs"
+	"golang.org/x/exp/mmap"
+)
+
+const (
+	// Bitcoin Chain Type: 4 bytes.
+	BitcoinChainTypeSize = 4
+
+	// Header Type: 1 byte.
+	HeaderTypeSize = 1
+
+	// Start Header Height: 4 bytes.
+	StartHeaderHeightSize = 4
+
+	// Header Metadata size in bytes (sum of all above): 9 bytes.
+	//
+	//nolint:lll
+	HeaderMetadataSize = BitcoinChainTypeSize + HeaderTypeSize + StartHeaderHeightSize
+
+	// Unknown HeaderSize.
+	UnknownHeaderSize = 0
+
+	// DefaultWriteBatchSizePerRegion defines the default number of
+	// headers to process in a single batch when no specific batch
+	// size is provided.
+	DEFAULT_WRITE_BATCH_SIZE_PER_REGION = 16384
```

Comment:
`DefaultWriteBatchSizePerRegion` would actually be the correct name as per Golang naming suggestions.